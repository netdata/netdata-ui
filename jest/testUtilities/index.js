import "@testing-library/jest-dom/extend-expect"

export { default as renderWithProviders } from "./renderWithProviders"
export { default as renderHookWithProviders } from "./renderHookWithProviders"

import userEvent from "@testing-library/user-event"
export const user = () => userEvent.setup()
export { userEvent }

export * from "@testing-library/react"
