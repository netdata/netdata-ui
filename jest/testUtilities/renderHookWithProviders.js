import React from "react"
import { renderHook } from "."
import { ThemeProvider } from "styled-components"
import { DefaultTheme } from "@/theme/default"

export default (hook, { theme = DefaultTheme, ...rest } = {}) =>
  renderHook(hook, {
    wrapper: props => <ThemeProvider theme={theme} {...props} />,
    ...rest,
  })
