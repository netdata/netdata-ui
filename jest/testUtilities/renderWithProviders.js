import React from "react"
import { render } from "."
import { ThemeProvider } from "styled-components"
import { DefaultTheme } from "@/theme/default"

export default (Component, { theme = DefaultTheme, ...rest } = {}) =>
  render(Component, {
    wrapper: props => <ThemeProvider theme={theme} {...props} />,
    ...rest,
  })
