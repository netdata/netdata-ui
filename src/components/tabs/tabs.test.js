/**
 * @jest-environment jsdom
 */

import React, { useState } from "react"
import { renderWithProviders, fireEvent } from "testUtilities"
import { Tabs, Tab } from "."

describe("Tabs states", () => {
  it(" * should render uncontrolled", () => {
    const { container, getByText, queryByText } = renderWithProviders(
      <Tabs>
        <Tab label="hi">Hello</Tab>
        <Tab label="Hi again">Hello again</Tab>
        <Tab label="Bye">Goodbye</Tab>
        <Tab label="Bye Bye">Fairwell</Tab>
      </Tabs>
    )

    const tabs = container.firstChild
    expect(tabs).toMatchSnapshot()

    expect(getByText("hi")).toHaveStyleRule("font-weight", "bold")
    expect(queryByText("Hello")).toBeInTheDocument()
    expect(queryByText("Hello again")).not.toBeInTheDocument()
    expect(queryByText("Goodbye")).not.toBeInTheDocument()
    expect(queryByText("Fairwell")).not.toBeInTheDocument()

    const tab = getByText("Bye Bye")
    fireEvent.click(tab)
    expect(tab).toHaveStyleRule("font-weight", "bold")
    expect(queryByText("Hello")).not.toBeInTheDocument()
    expect(queryByText("Fairwell")).toBeInTheDocument()
  })

  it(" * should render controlled", () => {
    const onChange = jest.fn()
    const Component = () => {
      const [selected, setSelected] = useState(2)

      return (
        <Tabs
          selected={selected}
          onChange={onChange.mockImplementation(index => setSelected(index))}
        >
          <Tab label="hi">Hello</Tab>
          <Tab label="Hi again">Hello again</Tab>
          <Tab label="Bye">Goodbye</Tab>
          <Tab label="Bye Bye">Fairwell</Tab>
        </Tabs>
      )
    }

    const { queryByText, getByText } = renderWithProviders(<Component />)

    expect(queryByText("Hello")).not.toBeInTheDocument()
    expect(queryByText("Hello again")).not.toBeInTheDocument()
    expect(getByText("Bye")).toHaveStyleRule("font-weight", "bold")
    expect(queryByText("Goodbye")).toBeInTheDocument()
    expect(queryByText("Fairwell")).not.toBeInTheDocument()

    const tab = getByText("Hi again")
    fireEvent.click(tab)
    expect(tab).toHaveStyleRule("font-weight", "bold")
    expect(onChange).toBeCalledWith(1)
    expect(queryByText("Goodbye")).not.toBeInTheDocument()
    expect(queryByText("Hello again")).toBeInTheDocument()
  })

  it(" * should not allow disabled to be selected", () => {
    const Component = () => (
      <Tabs>
        <Tab label="hi" disabled>
          Hello
        </Tab>
        <Tab label="Hi again">Hello again</Tab>
        <Tab label="Bye">Goodbye</Tab>
        <Tab label="Bye Bye">Fairwell</Tab>
      </Tabs>
    )
    const { queryByText, getByText } = renderWithProviders(<Component />)

    expect(queryByText("Hello")).not.toBeInTheDocument()
    expect(getByText("Hi again")).toHaveStyleRule("font-weight", "bold")
    expect(queryByText("Hello again")).toBeInTheDocument()

    const tab = getByText("hi")
    fireEvent.click(tab)
    expect(tab).toHaveStyleRule("font-weight", "normal")
    expect(queryByText("Hello")).not.toBeInTheDocument()
    expect(queryByText("Hello again")).toBeInTheDocument()
  })
})
