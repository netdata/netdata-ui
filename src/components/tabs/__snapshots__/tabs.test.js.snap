// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs states  * should render uncontrolled 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  color: #526161;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  padding: 0 2px;
  color: #526161;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #526161;
}

.c3 {
  white-space: nowrap;
  border-bottom: 4px solid #42B861;
  box-sizing: border-box;
  min-width: 80px;
  max-width: 208px;
  height: 48px;
  color: #526161;
  font-weight: bold;
  cursor: pointer;
  opacity: 1;
  pointer-events: auto;
}

.c3:hover {
  border-bottom: 4px solid #00AB44;
}

.c3 > span {
  font-weight: bold;
}

.c4 {
  white-space: nowrap;
  border-bottom: 4px solid rgba(255,255,255,0.0);
  box-sizing: border-box;
  min-width: 80px;
  max-width: 208px;
  height: 48px;
  color: #526161;
  font-weight: normal;
  cursor: pointer;
  opacity: 1;
  pointer-events: auto;
}

.c4:hover {
  border-bottom: 4px solid #00AB44;
}

.c4 > span {
  font-weight: normal;
}

<div
  class="c0"
>
  <nav
    class="c1 tabs tabs"
  >
    <div
      class="c2 c3"
      label="hi"
    >
      hi
    </div>
    <div
      class="c2 c4"
      label="Hi again"
    >
      Hi again
    </div>
    <div
      class="c2 c4"
      label="Bye"
    >
      Bye
    </div>
    <div
      class="c2 c4"
      label="Bye Bye"
    >
      Bye Bye
    </div>
  </nav>
  Hello
</div>
`;
